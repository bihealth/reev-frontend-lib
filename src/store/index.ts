import { createPinia } from 'pinia'

export * from './types'

export const pinia = createPinia()
