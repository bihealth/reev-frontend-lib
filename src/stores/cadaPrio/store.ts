/**
 * Store for genes ranking from cada-prio.
 *
 * This includes the data retrieved from the APIs.
 */
import { defineStore } from 'pinia'
import { ref } from 'vue'

import { StoreState } from '..'
import { CadaPrioClient } from '../../api/cadaPrio'
import { GeneRank } from './types'

/** Options for `loadData()` */
export interface LoadDataOptions {
  /** HGNC IDs to use for querying. */
  hgncIds?: string[]
  /** Whether to skip Clearing */
  skipClear?: boolean
}

export const useCadaPrioStore = defineStore('cadaPrio', () => {
  /** The current store state. */
  const storeState = ref<StoreState>(StoreState.Initial)

  /** The current gene Ranking. */
  const geneRanking = ref<GeneRank[] | undefined>(undefined)

  /**
   * Clear all data from store and reset it into initial state.
   */
  function clearData() {
    storeState.value = StoreState.Initial
    geneRanking.value = undefined
  }

  /**
   * Run prioritization and return results for the given HPO terms and HGNC IDs.
   *
   * @param hpoTerms HPO terms to use for querying.
   * @param options Options for the query.
   */
  const loadData = async (hpoTerms: string[], options?: LoadDataOptions) => {
    // Clear against artifact unless skipped.
    if (!options?.skipClear) {
      clearData()
    } else {
      // also clear if hpoTerms is empty, then exit
      if (hpoTerms.length === 0) {
        clearData()
        return
      }
    }

    // Load data via API
    storeState.value = StoreState.Loading
    try {
      const client = new CadaPrioClient()
      const response = await client.predictGeneImpact(hpoTerms, options?.hgncIds)
      const data = response.entries
      if (data.length === 0) {
        geneRanking.value = undefined
      } else {
        geneRanking.value = response.entries
      }
      storeState.value = StoreState.Active
    } catch (error) {
      console.error(error)
      storeState.value = StoreState.Error
    }
  }

  return {
    storeState,
    geneRanking,
    loadData,
    clearData
  }
})
