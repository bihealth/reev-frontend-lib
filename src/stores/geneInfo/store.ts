/**
 * Store for gene details.
 *
 * This includes the data retrieved from the APIs.
 */
import { defineStore } from 'pinia'
import { ref } from 'vue'

import { AnnonarsClient } from '../../api/annonars'
import { MehariClient } from '../../api/mehari'
import { type HpoTerm, VigunoClient } from '../../api/viguno'
import { type GenomeBuild } from '../../lib/genomeBuilds'
import { ClinvarPerGeneRecord } from '../../pbs/annonars/clinvar/per_gene'
import { Record as GeneInfoRecord } from '../../pbs/annonars/genes/base'
import { GenomeBuild as MehariGenomeBuild, Transcript } from '../../pbs/mehari/txs'
import { StoreState } from '../types'

export const useGeneInfoStore = defineStore('geneInfo', () => {
  /** The current store state. */
  const storeState = ref<StoreState>(StoreState.Initial)

  /** The current gene query. */
  const hgncId = ref<string | undefined>(undefined)

  /** The retrieved gene data from annonars. */
  const geneInfo = ref<GeneInfoRecord | undefined>(undefined)

  /** The HPO terms retrieved from viguno. */
  const hpoTerms = ref<HpoTerm[]>([])

  /** ClinVar gene-related information from annoars. */
  const geneClinvar = ref<ClinvarPerGeneRecord | undefined>(undefined)

  /** Transcript information from dotty (unless dotty API not available). */
  const transcripts = ref<Transcript[] | undefined>(undefined)

  function clearData() {
    storeState.value = StoreState.Initial
    hgncId.value = undefined
    geneInfo.value = undefined
    geneClinvar.value = undefined
    transcripts.value = undefined
  }

  const initialize = async (hgncIdQuery: string, genomeBuild: GenomeBuild) => {
    // Do not re-load data if the gene symbol is the same
    if (hgncIdQuery === hgncId.value) {
      return
    }

    // Clear against artifact
    clearData()

    // Load data via API
    storeState.value = StoreState.Loading
    try {
      const annonarsClient = new AnnonarsClient()
      const data = await annonarsClient.fetchGeneInfo(hgncIdQuery)
      geneInfo.value = undefined
      for (const gene of data.genes) {
        if (gene.hgnc!.hgncId === hgncIdQuery) {
          geneInfo.value = gene
          break
        }
      }
      if (geneInfo.value === undefined) {
        throw new Error(`No gene data found for HGNC ID ${hgncIdQuery}`)
      }

      try {
        geneClinvar.value = await annonarsClient.fetchGeneClinvarInfo(hgncIdQuery)
      } catch (e) {
        throw new Error(`No gene clinvar data found: ${e}`)
      }

      const vigunoClient = new VigunoClient()
      const hpoTermsData = await vigunoClient.fetchHpoTermsForHgncId(hgncIdQuery)
      if (hpoTermsData === null) {
        throw new Error('Problem querying HPO terms data.')
      }
      hpoTerms.value = hpoTermsData.result

      hgncId.value = hgncIdQuery
      storeState.value = StoreState.Active

      const mehariClient = new MehariClient()
      const transcriptsData = await mehariClient.retrieveGeneTranscripts(
        hgncIdQuery,
        genomeBuild === 'grch37'
          ? MehariGenomeBuild.GENOME_BUILD_GRCH37
          : MehariGenomeBuild.GENOME_BUILD_GRCH38
      )
      transcripts.value = transcriptsData.transcripts ?? undefined
    } catch (e) {
      console.error('There was an error loading the gene data.', e)
      clearData()
      storeState.value = StoreState.Error
    }
  }

  return {
    storeState,
    hgncId,
    geneInfo,
    geneClinvar,
    hpoTerms,
    transcripts,
    initialize,
    clearData
  }
})
