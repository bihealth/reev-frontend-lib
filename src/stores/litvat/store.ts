/**
 * Store wrapping the LitVar access.
 */
import { defineStore } from 'pinia'
import { ref } from 'vue'

import { LitVarClient } from '../../api/litvar'
import { StoreState } from '../types'
import { type SearchResults } from './types'

export const useLitVarStore = defineStore('litvar', () => {
  /** The current store state. */
  const storeState = ref<StoreState>(StoreState.Initial)

  /** The Seqvar name currently loaded for. */
  const seqVar = ref<string | undefined>(undefined)

  /** Detailed result information. */
  const searchResults = ref<SearchResults>({})

  /** Initialize the store for the given SeqVar name. */
  const initialize = async (seqVar$?: string, force: boolean = false) => {
    // Skip if already loaded
    if (!force && seqVar$ === seqVar.value) {
      return
    }

    // Clear against artifacts.
    clearData()

    storeState.value = StoreState.Loading

    // Bail out if no SeqVar name is available.
    if (!seqVar$) {
      storeState.value = StoreState.Active
      return
    }

    // "Just" lookup via LitVar client.
    const client = new LitVarClient()
    try {
      searchResults.value = await client.performSearch(seqVar$)
      storeState.value = StoreState.Active
    } catch (err) {
      storeState.value = StoreState.Error
      throw new Error(`Error running LitVar search: ${err}`)
    }
  }

  /** Clear the store. */
  const clearData = () => {
    storeState.value = StoreState.Initial
    seqVar.value = undefined
    searchResults.value = {}
  }

  /** The current gene Ranking. */
  return {
    storeState,
    seqVar,
    searchResults,
    initialize,
    clearData
  }
})