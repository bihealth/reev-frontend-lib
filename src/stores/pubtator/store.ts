/**
 * Store wrapping the PubTator access.
 */
import { defineStore } from 'pinia'
import { ref } from 'vue'

import { PubtatorClient } from '../../api/pubtator'
import { StoreState } from '../types'
import { type SearchResults } from './types'

export const usePubtatorStore = defineStore('pubtator', () => {
  /** The current store state. */
  const storeState = ref<StoreState>(StoreState.Initial)

  /** The HGNC symbol currently loaded for. */
  const hgncSymbol = ref<string | undefined>(undefined)

  /** Detailed result information. */
  const searchResults = ref<SearchResults>({})

  /**
   * Initialize the store for the given HGNC symbol.
   *
   * @param hgncSymbol$ HGNC symbol to initialize the store for.
   * @param force Whether to force re-initialization.
   * @throws Error if the search fails.
   */
  const initialize = async (hgncSymbol$?: string, force: boolean = false) => {
    // Skip if already loaded
    if (!force && hgncSymbol$ === hgncSymbol.value) {
      return
    }

    // Clear against artifacts.
    clearData()

    storeState.value = StoreState.Loading

    // Bail out if no HGNC symbol is available.
    if (!hgncSymbol$) {
      storeState.value = StoreState.Active
      return
    }

    // "Just" lookup via PubTator client.
    const client = new PubtatorClient()
    try {
      searchResults.value = await client.performSearch(hgncSymbol$)
      storeState.value = StoreState.Active
    } catch (err) {
      storeState.value = StoreState.Error
      throw new Error(`Error running PubTator 3 search: ${err}`)
    }
  }

  /** Clear the store. */
  const clearData = () => {
    storeState.value = StoreState.Initial
    hgncSymbol.value = undefined
    searchResults.value = {}
  }

  /** The current gene Ranking. */
  return {
    storeState,
    hgncSymbol,
    searchResults,
    initialize,
    clearData
  }
})
