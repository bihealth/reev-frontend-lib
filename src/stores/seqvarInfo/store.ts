/**
 * Store for variant details.
 *
 * This includes the data retrieved from the APIs.
 */
import equal from 'fast-deep-equal'
import { defineStore } from 'pinia'
import { ref } from 'vue'

import { StoreState } from '..'
import { AnnonarsClient, SeqvarInfoResult } from '../../api/annonars'
import { MehariClient, SeqvarResultEntry } from '../../api/mehari'
import { type HpoTerm, VigunoClient } from '../../api/viguno'
import { type Seqvar } from '../../lib/genomicVars'
import { ClinvarPerGeneRecord } from '../../pbs/annonars/clinvar/per_gene'
import { Record as GeneInfoRecord } from '../../pbs/annonars/genes/base'

export const useSeqvarInfoStore = defineStore('seqvarInfo', () => {
  /** The current store state. */
  const storeState = ref<StoreState>(StoreState.Initial)

  /** The Seqvar that the previous record has been retrieved for. */
  const seqvar = ref<Seqvar | undefined>(undefined)

  /** The optional HGNC ID to limit the query to. */
  const hgncId = ref<string | undefined>(undefined)

  /** Variant-related information from annonars. */
  const varAnnos = ref<SeqvarInfoResult | undefined>(undefined)

  /** ClinVar gene-related information from annoars. */
  const geneClinvar = ref<ClinvarPerGeneRecord | undefined>(undefined)

  /** Information about related gene. */
  const geneInfo = ref<GeneInfoRecord | undefined>(undefined)

  /** The HPO terms retrieved from viguno. */
  const hpoTerms = ref<HpoTerm[]>([])

  /** Transcript consequence information from mehari. */
  const txCsq = ref<SeqvarResultEntry[] | undefined>(undefined)

  /** Promise for initialization of the store. */
  const initializeRes = ref<Promise<any> | undefined>(undefined)

  /** Clear all data in the store. */
  const clearData = () => {
    storeState.value = StoreState.Initial
    seqvar.value = undefined
    hgncId.value = undefined
    varAnnos.value = undefined
    txCsq.value = undefined
    geneInfo.value = undefined
    geneClinvar.value = undefined
  }

  /**
   * Initialize and oad data from the server.
   *
   * @param seqvar$ The sequence variant to use for the query.
   * @param hgncId$ Optional HGNC ID to limit the query to.
   * @param forceReload Whether to force-reload in case the variant is the same.
   * @returns
   */
  const initialize = async (seqvar$: Seqvar, hgncId$?: string, forceReload: boolean = false) => {
    // Protect against loading multiple times.
    if (
      !forceReload &&
      storeState.value !== StoreState.Initial &&
      equal(seqvar$, seqvar.value) &&
      hgncId$ === hgncId.value
    ) {
      return initializeRes.value
    }

    // Clear against artifact
    clearData()

    // Load data via API
    storeState.value = StoreState.Loading
    const annonarsClient = new AnnonarsClient()
    const mehariClient = new MehariClient()
    const vigunoClient = new VigunoClient()

    // Retrieve variant information from annonars and mehari.
    initializeRes.value = Promise.all([
      annonarsClient.fetchVariantInfo(seqvar$).then((data) => {
        varAnnos.value = data.result
      }),
      mehariClient.retrieveSeqvarsCsq(seqvar$, hgncId$).then((data) => {
        txCsq.value = data.result ?? []
      })
    ])
      .then((): Promise<any> => {
        if (txCsq.value !== undefined && txCsq.value.length !== 0) {
          const localHgncId = txCsq.value[0].geneId
          return Promise.all([
            annonarsClient.fetchGeneInfo(localHgncId).then((data) => {
              for (const gene of data.genes) {
                if (gene.hgnc!.hgncId === localHgncId) {
                  geneInfo.value = gene
                }
              }
            }),
            annonarsClient.fetchGeneClinvarInfo(localHgncId).then((data) => {
              geneClinvar.value = data
            }),
            // vigunoClient.fetchHpoTermsForHgncId(localHgncId).then((data) => {
            //   hpoTerms.value = data.result
            // })
          ])
        } else {
          return Promise.resolve()
        }
      })
      .then(() => {
        hgncId.value = hgncId$
        seqvar.value = seqvar$
        storeState.value = StoreState.Active
      })
      .catch((err) => {
        console.error('There was an error loading the variant data.', err)
        clearData()
        storeState.value = StoreState.Error
      })

    return initializeRes.value
  }

  return {
    initializeRes,
    storeState,
    seqvar,
    varAnnos,
    geneClinvar,
    geneInfo,
    hpoTerms,
    txCsq,
    initialize,
    clearData
  }
})
