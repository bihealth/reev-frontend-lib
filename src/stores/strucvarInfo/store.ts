/**
 * Store for variant details.
 *
 * This includes the data retrieved from the APIs.
 */
import equal from 'fast-deep-equal'
import { defineStore } from 'pinia'
import { ref } from 'vue'

import { StoreState } from '..'
import { AnnonarsClient } from '../../api/annonars'
import { MehariClient } from '../../api/mehari'
import { GeneTranscriptEffects } from '../../api/mehari/types'
import { type Strucvar } from '../../lib/genomicVars'
import { ResponseRecord as ClinvarStrucvarResponseRecord } from '../../pbs/annonars/clinvar/sv'
import { Record as GeneInfoRecord } from '../../pbs/annonars/genes/base'

export const useStrucvarInfoStore = defineStore('strucvarInfo', () => {
  /** The current store state. */
  const storeState = ref<StoreState>(StoreState.Initial)

  /** The current SV record. */
  const strucvar = ref<Strucvar | undefined>(undefined)

  /** The consequences of the `currentStrucvar` */
  const csq = ref<GeneTranscriptEffects[] | undefined>(undefined)

  /** Infos on the variants of the record. */
  const genesInfos = ref<GeneInfoRecord[] | undefined>(undefined)

  /** The ClinVar SV records. */
  const clinvarSvRecords = ref<ClinvarStrucvarResponseRecord[] | undefined>(undefined)

  function clearData() {
    storeState.value = StoreState.Initial
    csq.value = undefined
    strucvar.value = undefined
    genesInfos.value = undefined
  }

  /**
   * Initialize and load data from the server.
   *
   * @param strucvar$ The structural variant to use for the query.
   * @param forceReload Whether to force-reload in case the variant is the same.
   */
  const initialize = async (strucvar$: Strucvar, forceReload: boolean = false) => {
    // Protect against loading multiple times.
    if (
      !forceReload &&
      storeState.value !== StoreState.Initial &&
      equal(strucvar$, strucvar.value)
    ) {
      return
    }

    // Clear old store contents
    clearData()

    // Load data via API
    storeState.value = StoreState.Loading
    const annonarsClient = new AnnonarsClient()
    const mehariClient = new MehariClient()
    try {
      // Fetch SV record
      const mehariResult = await mehariClient.retrieveStrucvarsCsq(strucvar$)
      // Fetch ClinVar SV records
      const { records: responseRecords } = await annonarsClient.fetchClinvarStrucvars(strucvar$)
      clinvarSvRecords.value = responseRecords ?? []

      // Fetch gene infos
      const hgncIds = []
      for (const txEffect of mehariResult.result) {
        if (txEffect.hgncId) {
          hgncIds.push(txEffect.hgncId)
        }
      }
      if (hgncIds.length) {
        genesInfos.value = await annonarsClient.fetchGeneInfos(hgncIds)
      } else {
        genesInfos.value = []
      }

      // Sort by gene symbol
      genesInfos.value!.sort((a, b) =>
        (a?.hgnc!.hgncId ?? 'ZZZ').localeCompare(b?.hgnc!.hgncId ?? 'ZZZ')
      )

      csq.value = mehariResult.result
      strucvar.value = strucvar$
      storeState.value = StoreState.Active
    } catch (e) {
      console.error('There was an error loading the SV data.', e)
      storeState.value = StoreState.Error
    }
  }

  return {
    storeState,
    strucvar,
    csq,
    genesInfos,
    clinvarSvRecords,
    clearData,
    initialize
  }
})
