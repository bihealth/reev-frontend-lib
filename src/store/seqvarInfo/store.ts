/**
 * Store for variant details.
 *
 * This includes the data retrieved from the APIs.
 */
import equal from 'fast-deep-equal'
import { defineStore } from 'pinia'
import { ref } from 'vue'

import { AnnonarsClient, SeqvarInfoResult } from '../../api/annonars'
import { MehariClient, SeqvarResultEntry } from '../../api/mehari'
import { type HpoTerm, VigunoClient } from '../../api/viguno'
import { type Seqvar } from '../../lib/genomicVars'
import { ClinvarPerGeneRecord } from '../../pbs/annonars/clinvar/per_gene'
import { Record as GeneInfoRecord } from '../../pbs/annonars/genes/base'
import { StoreState } from '../../store'

export const useSeqvarInfoStore = defineStore('seqvarInfo', () => {
  /** The current store state. */
  const storeState = ref<StoreState>(StoreState.Initial)

  /** The Seqvar that the previous record has been retrieved for. */
  const seqvar = ref<Seqvar | undefined>(undefined)

  /** Variant-related information from annonars. */
  const varAnnos = ref<SeqvarInfoResult | undefined>(undefined)

  /** ClinVar gene-related information from annoars. */
  const geneClinvar = ref<ClinvarPerGeneRecord | undefined>(undefined)

  /** Information about related gene. */
  const geneInfo = ref<GeneInfoRecord | undefined>(undefined)

  /** The HPO terms retrieved from viguno. */
  const hpoTerms = ref<HpoTerm[]>([])

  /** Transcript consequence information from mehari. */
  const txCsq = ref<SeqvarResultEntry[] | undefined>(undefined)

  /** Promise for initialization of the store. */
  const initializeRes = ref<Promise<any> | undefined>(undefined)

  /** Clear all data in the store. */
  const clearData = () => {
    storeState.value = StoreState.Initial
    seqvar.value = undefined
    varAnnos.value = undefined
    txCsq.value = undefined
    geneInfo.value = undefined
    geneClinvar.value = undefined
  }

  /**
   * Initialize and oad data from the server.
   *
   * @param seqvar$ The sequence variant to use for the query.
   * @param forceReload Whether to force-reload in case the variant is the same.
   * @returns
   */
  const initialize = async (seqvar$: Seqvar, forceReload: boolean = false) => {
    // Protect against loading multiple times.
    if (!forceReload && storeState.value !== StoreState.Initial && equal(seqvar$, seqvar.value)) {
      return initializeRes.value
    }

    // Clear against artifact
    clearData()

    // Load data via API
    storeState.value = StoreState.Loading
    const annonarsClient = new AnnonarsClient()
    const mehariClient = new MehariClient()
    const vigunoClient = new VigunoClient()
    let hgncId = ''

    // Retrieve variant information from annonars and mehari.
    initializeRes.value = Promise.all([
      annonarsClient.fetchVariantInfo(seqvar$).then((data) => {
        varAnnos.value = data.result
      }),
      mehariClient.retrieveSeqvarsCsq(seqvar$).then((data) => {
        txCsq.value = data.result ?? []
      })
    ])
      .then((): Promise<any> => {
        if (txCsq.value !== undefined && txCsq.value.length !== 0) {
          hgncId = txCsq.value[0].geneId
          return Promise.all([
            annonarsClient.fetchGeneInfo(hgncId).then((data) => {
              for (const gene of data.genes) {
                if (gene.hgnc!.hgncId === hgncId) {
                  geneInfo.value = gene
                }
              }
            }),
            annonarsClient.fetchGeneClinvarInfo(hgncId).then((data) => {
              geneClinvar.value = data
            }),
            vigunoClient.fetchHpoTermsForHgncId(hgncId).then((data) => {
              if (!data.results.length) {
                throw new Error('No HPO terms found.')
              }
              hpoTerms.value = data.results
            })
          ])
        } else {
          return Promise.resolve()
        }
      })
      .then(() => {
        seqvar.value = seqvar$
        storeState.value = StoreState.Active
      })
      .catch((err) => {
        console.error('There was an error loading the variant data.', err)
        clearData()
        storeState.value = StoreState.Error
      })

    return initializeRes.value
  }

  return {
    initializeRes,
    storeState,
    seqvar,
    varAnnos,
    geneClinvar,
    geneInfo,
    hpoTerms,
    txCsq,
    initialize,
    clearData
  }
})
