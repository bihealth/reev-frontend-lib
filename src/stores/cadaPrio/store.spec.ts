import fs from 'fs'
import path from 'path'
import { createPinia, setActivePinia } from 'pinia'
import { beforeEach, describe, expect, it, vi } from 'vitest'
import createFetchMock from 'vitest-fetch-mock'

import { StoreState } from '..'
import { useCadaPrioStore } from './store'

const predictResponseJson = JSON.parse(
  fs.readFileSync(
    path.resolve(__dirname, '../../api/cadaPrio/fixture.predictResponse.json'),
    'utf8'
  )
)

const fetchMocker = createFetchMock(vi)

describe('Cada Prio Store', () => {
  beforeEach(() => {
    setActivePinia(createPinia())
    fetchMocker.enableMocks()
    fetchMocker.resetMocks()
  })

  it('should have initial state', () => {
    // arrange:
    const store = useCadaPrioStore()

    // act: nothing to do

    // assert:
    expect(store.storeState).toEqual(StoreState.Initial)
    expect(store.geneRanking).toBe(undefined)
  })

  it('should predict gene impact', async () => {
    // arrange:
    fetchMocker.mockResponse(JSON.stringify(predictResponseJson))
    const store = useCadaPrioStore()

    // act:
    await store.loadData(['HP:0000001'])

    // assert:
    expect(store.storeState).toEqual(StoreState.Active)
    expect(store.geneRanking).toMatchSnapshot()
  })

  it('should handle error when predicting gene impact', async () => {
    // arrange:
    // Disable error logging
    vi.spyOn(console, 'error').mockImplementation(() => {})
    fetchMocker.mockReject(new Error('Internal Server Error'))
    const store = useCadaPrioStore()

    // act:
    await store.loadData(['HP:0000001'])

    // assert:
    expect(store.storeState).toEqual(StoreState.Error)
    expect(store.geneRanking).toBe(undefined)
  })
})
