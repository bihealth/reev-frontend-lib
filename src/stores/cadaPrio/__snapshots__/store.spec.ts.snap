// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Cada Prio Store > should predict gene impact 1`] = `
[
  {
    "geneSymbol": "WNT7A",
    "hgncId": "HGNC:12786",
    "ncbiGeneId": "7476",
    "rank": 1,
    "score": 76.84823608398438,
  },
  {
    "geneSymbol": "SMARCA2",
    "hgncId": "HGNC:11098",
    "ncbiGeneId": "6595",
    "rank": 2,
    "score": 76.46592712402344,
  },
  {
    "geneSymbol": "EFL1",
    "hgncId": "HGNC:25789",
    "ncbiGeneId": "79631",
    "rank": 3,
    "score": 69.20905303955078,
  },
  {
    "geneSymbol": "SRP54",
    "hgncId": "HGNC:11301",
    "ncbiGeneId": "6729",
    "rank": 4,
    "score": 65.58743286132812,
  },
  {
    "geneSymbol": "HPGD",
    "hgncId": "HGNC:5154",
    "ncbiGeneId": "3248",
    "rank": 5,
    "score": 63.857845306396484,
  },
  {
    "geneSymbol": "GALNS",
    "hgncId": "HGNC:4122",
    "ncbiGeneId": "2588",
    "rank": 6,
    "score": 63.840911865234375,
  },
  {
    "geneSymbol": "DNAJC21",
    "hgncId": "HGNC:27030",
    "ncbiGeneId": "134218",
    "rank": 7,
    "score": 63.35689926147461,
  },
  {
    "geneSymbol": "NOG",
    "hgncId": "HGNC:7866",
    "ncbiGeneId": "9241",
    "rank": 8,
    "score": 63.207393646240234,
  },
  {
    "geneSymbol": "ESCO2",
    "hgncId": "HGNC:27230",
    "ncbiGeneId": "157570",
    "rank": 9,
    "score": 63.09525680541992,
  },
  {
    "geneSymbol": "COL11A1",
    "hgncId": "HGNC:2186",
    "ncbiGeneId": "1301",
    "rank": 10,
    "score": 61.24506378173828,
  },
  {
    "geneSymbol": "PTDSS1",
    "hgncId": "HGNC:9587",
    "ncbiGeneId": "9791",
    "rank": 11,
    "score": 60.995365142822266,
  },
  {
    "geneSymbol": "VAC14",
    "hgncId": "HGNC:25507",
    "ncbiGeneId": "55697",
    "rank": 12,
    "score": 60.21630859375,
  },
  {
    "geneSymbol": "TBX3",
    "hgncId": "HGNC:11602",
    "ncbiGeneId": "6926",
    "rank": 13,
    "score": 59.911136627197266,
  },
  {
    "geneSymbol": "GRIA3",
    "hgncId": "HGNC:4573",
    "ncbiGeneId": "2892",
    "rank": 14,
    "score": 59.312957763671875,
  },
  {
    "geneSymbol": "RNU4ATAC",
    "hgncId": "HGNC:34016",
    "ncbiGeneId": "100151683",
    "rank": 15,
    "score": 59.29935836791992,
  },
  {
    "geneSymbol": "GDF5",
    "hgncId": "HGNC:4220",
    "ncbiGeneId": "8200",
    "rank": 16,
    "score": 58.710994720458984,
  },
  {
    "geneSymbol": "TRAPPC2",
    "hgncId": "HGNC:23068",
    "ncbiGeneId": "6399",
    "rank": 17,
    "score": 56.57151412963867,
  },
  {
    "geneSymbol": "IHH",
    "hgncId": "HGNC:5956",
    "ncbiGeneId": "3549",
    "rank": 18,
    "score": 55.67450714111328,
  },
  {
    "geneSymbol": "PCNT",
    "hgncId": "HGNC:16068",
    "ncbiGeneId": "5116",
    "rank": 19,
    "score": 55.3505859375,
  },
]
`;
